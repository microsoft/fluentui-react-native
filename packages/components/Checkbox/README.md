# Checkbox

A cross-platform Checkbox component using the Fluent Design System.

Supported Platforms: Android, iOS, macOS, web, windows, win32

Note that on iOS, the more common control to use is a Switch
https://developer.apple.com/design/human-interface-guidelines/ios/controls/switches/
https://reactnative.dev/docs/switch
