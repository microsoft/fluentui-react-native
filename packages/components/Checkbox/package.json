{
  "name": "@fluentui-react-native/checkbox",
  "version": "0.11.8",
  "description": "A cross-platform Checkbox component using the Fluent Design System",
  "main": "src/index.ts",
  "module": "src/index.ts",
  "typings": "lib/index.d.ts",
  "onPublish": {
    "main": "lib-commonjs/index.js",
    "module": "lib/index.js"
  },
  "scripts": {
    "build": "fluentui-scripts build",
    "clean": "fluentui-scripts clean",
    "depcheck": "fluentui-scripts depcheck",
    "just": "fluentui-scripts",
    "lint": "fluentui-scripts eslint",
    "test": "fluentui-scripts jest",
    "update-snapshots": "fluentui-scripts jest -u",
    "verify-api": "fluentui-scripts verify-api-extractor",
    "update-api": "fluentui-scripts update-api-extractor",
    "prettier": "fluentui-scripts prettier",
    "prettier-fix": "fluentui-scripts prettier --fix true"
  },
  "dependencies": {
    "@uifabricshared/foundation-compose": "^1.9.4",
    "@fluentui-react-native/adapters": ">=0.6.1 <1.0.0",
    "@fluentui-react-native/interactive-hooks": ">=0.10.8 <1.0.0",
    "@fluentui-react-native/text": ">=0.9.6 <1.0.0",
    "@fluentui-react-native/tokens": ">=0.9.3 <1.0.0",
    "@uifabricshared/foundation-composable": ">=0.8.2 <1.0.0",
    "@uifabricshared/foundation-settings": ">=0.9.1 <1.0.0"
  },
  "devDependencies": {
    "@office-iss/react-native-win32": "^0.63.7",
    "@types/react-native": "^0.63.0",
    "@uifabricshared/build-native": "^0.1.1",
    "@uifabricshared/eslint-config-rules": "^0.1.1",
    "react-native": "^0.63.4"
  },
  "peerDependencies": {
    "react": ">=16.13.1",
    "react-native": ">=0.63.4"
  },
  "author": "",
  "license": "MIT"
}
