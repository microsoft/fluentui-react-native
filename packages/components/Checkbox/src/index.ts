export * from './Checkbox.types';
export * from './Checkbox';
