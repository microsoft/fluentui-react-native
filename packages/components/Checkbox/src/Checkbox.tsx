/** @jsx withSlots */
import * as React from 'react';
import { View } from 'react-native';
import { Text } from '@fluentui-react-native/text';
import { ICheckboxState, ICheckboxProps, ICheckboxSlotProps, ICheckboxRenderData, ICheckboxType, checkboxName } from './Checkbox.types';
import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
import { filterViewProps } from '@fluentui-react-native/adapters';
import { settings, checkboxSelectActionLabel } from './Checkbox.settings';
import { mergeSettings } from '@uifabricshared/foundation-settings';
import { foregroundColorTokens, textTokens, borderTokens, getPaletteFromTheme } from '@fluentui-react-native/tokens';
import {
  useAsToggle,
  useAsPressable,
  useViewCommandFocus,
  useKeyCallback,
  useOnPressWithFocus,
} from '@fluentui-react-native/interactive-hooks';
import { backgroundColorTokens } from '@fluentui-react-native/tokens';

export const Checkbox = compose<ICheckboxType>({
  displayName: checkboxName,

  usePrepareProps: (userProps: ICheckboxProps, useStyling: IUseComposeStyling<ICheckboxType>) => {
    const {
      ariaLabel,
      checked,
      defaultChecked,
      boxSide,
      disabled,
      label,
      onChange,
      componentRef = React.useRef(null),
      ...rest
    } = userProps;

    // Warns defaultChecked and checked being mutually exclusive.
    if (defaultChecked != undefined && checked != undefined) {
      console.warn('defaultChecked and checked are mutually exclusive to one another. Use one or the other.');
    }

    // Re-usable hook for toggle components.
    const [isChecked, toggleChecked] = useAsToggle(defaultChecked, checked, onChange);

    // Ensure focus is placed on checkbox after click
    const toggleCheckedWithFocus = useOnPressWithFocus(componentRef, toggleChecked);

    const pressable = useAsPressable({ onPress: toggleCheckedWithFocus, ...rest });

    const buttonRef = useViewCommandFocus(componentRef);

    // Handles the "Space" key toggling the Checkbox
    const onKeyUpSpace = useKeyCallback(toggleChecked, ' ');

    const state: ICheckboxState = {
      ...pressable.state,
      disabled: !!disabled,
      checked: isChecked,
      boxAtEnd: boxSide == undefined || boxSide == 'start' ? false : true,
    };

    // Grab the styling information from the userProps, referencing the state as well as the props.
    const styleProps = useStyling(userProps, (override: string) => state[override] || userProps[override]);

    // Used when creating accessibility properties in mergeSettings below
    const onAccessibilityAction = React.useCallback(
      (event: { nativeEvent: { actionName: any } }) => {
        switch (event.nativeEvent.actionName) {
          case 'Toggle':
            toggleChecked();
            break;
        }
      },
      [toggleChecked, userProps, state, pressable.props],
    );

    const slotProps = mergeSettings<ICheckboxSlotProps>(styleProps, {
      root: {
        rest,
        ref: buttonRef,
        ...pressable.props,
        accessibilityRole: 'checkbox',
        accessibilityLabel: ariaLabel || label,
        accessibilityState: { disabled: state.disabled, checked: state.checked },
        accessibilityActions: [{ name: 'Toggle', label: checkboxSelectActionLabel }],
        onAccessibilityAction: onAccessibilityAction,
        onKeyUp: onKeyUpSpace,
      },
      // Temporary checkmark until SVG functionality
      checkmark: { children: 'âœ“' },
      content: { children: label },
    });

    return { slotProps, state };
  },

  render: (Slots: ISlots<ICheckboxSlotProps>, renderData: ICheckboxRenderData, ...children: React.ReactNode[]) => {
    return (
      <Slots.root>
        {renderData?.state.boxAtEnd && <Slots.content />}
        <Slots.checkbox>
          <Slots.checkmark />
        </Slots.checkbox>
        {!renderData?.state.boxAtEnd && <Slots.content />}
        {children}
      </Slots.root>
    );
  },

  settings,
  slots: {
    root: View,
    checkbox: { slotType: View, filter: filterViewProps },
    checkmark: Text,
    content: Text,
  },
  styles: {
    root: [],
    checkbox: [
      backgroundColorTokens,
      borderTokens,
      [
        { source: 'checkboxBackgroundColor', lookup: getPaletteFromTheme, target: 'backgroundColor' },
        { source: 'checkboxBorderColor', lookup: getPaletteFromTheme, target: 'borderColor' },
      ],
    ],
    checkmark: [
      foregroundColorTokens,
      [
        { source: 'checkmarkColor', lookup: getPaletteFromTheme, target: 'color' },
        { source: 'checkmarkVisibility', target: 'opacity' },
      ],
    ],
    content: [foregroundColorTokens, textTokens, [{ source: 'textBorderColor', lookup: getPaletteFromTheme, target: 'borderColor' }]],
  },
});

export default Checkbox;
