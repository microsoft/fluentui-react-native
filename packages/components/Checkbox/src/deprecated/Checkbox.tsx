/** @jsxRuntime classic */
/** @jsx withSlots */
import * as React from 'react';
import { View } from 'react-native';

import { filterViewProps } from '@fluentui-react-native/adapters';
import {
  useAsToggle,
  useAsPressable,
  useViewCommandFocus,
  useKeyProps,
  useOnPressWithFocus,
} from '@fluentui-react-native/interactive-hooks';
import type { IPressableProps } from '@fluentui-react-native/pressable';
import { Text } from '@fluentui-react-native/text';
import { foregroundColorTokens, textTokens, borderTokens, getPaletteFromTheme } from '@fluentui-react-native/tokens';
import { backgroundColorTokens } from '@fluentui-react-native/tokens';
import type { ISlots } from '@uifabricshared/foundation-composable';
import { withSlots } from '@uifabricshared/foundation-composable';
import type { IUseComposeStyling } from '@uifabricshared/foundation-compose';
import { compose } from '@uifabricshared/foundation-compose';
import { mergeSettings } from '@uifabricshared/foundation-settings';

import { settings, checkboxSelectActionLabel } from './Checkbox.settings';
import type { ICheckboxState, ICheckboxProps, ICheckboxSlotProps, ICheckboxRenderData, ICheckboxType } from './Checkbox.types';
import { checkboxName } from './Checkbox.types';

/**
 * @deprecated This will be removed when the package moves to 1.0.0.
 * Please see MIGRATION.md for details on how to move to the new Checkbox.
 */
export const Checkbox = compose<ICheckboxType>({
  displayName: checkboxName,

  usePrepareProps: (userProps: ICheckboxProps, useStyling: IUseComposeStyling<ICheckboxType>) => {
    const defaultComponentRef = React.useRef(null);
    const {
      accessible,
      accessibilityLabel,
      accessibilityRole,
      ariaLabel,
      checked,
      defaultChecked,
      boxSide,
      label,
      onChange,
      componentRef = defaultComponentRef,
      focusable,
      ...rest
    } = userProps;

    // Warns defaultChecked and checked being mutually exclusive.
    if (defaultChecked != undefined && checked != undefined) {
      console.warn('defaultChecked and checked are mutually exclusive to one another. Use one or the other.');
    }

    // Re-usable hook for toggle components.
    const [isChecked, toggleChecked] = useAsToggle(defaultChecked, checked, onChange);

    // Ensure focus is placed on checkbox after click
    const toggleCheckedWithFocus = useOnPressWithFocus(componentRef, toggleChecked);

    const pressable = useAsPressable({ onPress: toggleCheckedWithFocus, ...(rest as IPressableProps) });

    const buttonRef = useViewCommandFocus(componentRef);

    // Handles the "Space" key toggling the Checkbox
    const onKeyProps = useKeyProps(toggleChecked, ' ');

    const state: ICheckboxState = {
      ...pressable.state,
      disabled: !!userProps.disabled,
      checked: isChecked,
      boxAtEnd: boxSide == undefined || boxSide == 'start' ? false : true,
    };

    // Grab the styling information from the userProps, referencing the state as well as the props.
    const styleProps = useStyling(userProps, (override: string) => state[override] || userProps[override]);

    // Used when creating accessibility properties in mergeSettings below
    const onAccessibilityAction = React.useCallback(
      (event: { nativeEvent: { actionName: any } }) => {
        switch (event.nativeEvent.actionName) {
          case 'Toggle':
            toggleChecked();
            break;
        }
      },
      [toggleChecked, userProps, state, pressable.props],
    );

    const slotProps = mergeSettings<ICheckboxSlotProps>(styleProps, {
      root: {
        rest,
        ref: buttonRef,
        ...pressable.props,
        accessible: accessible ?? true,
        accessibilityRole: accessibilityRole ?? 'checkbox',
        accessibilityLabel: accessibilityLabel ?? label,
        accessibilityState: { disabled: state.disabled, checked: state.checked },
        accessibilityActions: [{ name: 'Toggle', label: checkboxSelectActionLabel }],
        focusable: focusable ?? !state.disabled,
        onAccessibilityAction: onAccessibilityAction,
        enableFocusRing: false,
        ...onKeyProps,
      },
      // Temporary checkmark until SVG functionality
      checkmark: { children: 'âœ“' },
      content: { children: label },
    });

    return { slotProps, state };
  },

  render: (Slots: ISlots<ICheckboxSlotProps>, renderData: ICheckboxRenderData, ...children: React.ReactNode[]) => {
    return (
      <Slots.root>
        {renderData?.state.boxAtEnd && <Slots.content />}
        <Slots.checkbox>
          <Slots.checkmark />
        </Slots.checkbox>
        {!renderData?.state.boxAtEnd && <Slots.content />}
        {children}
      </Slots.root>
    );
  },

  settings,
  slots: {
    root: View,
    checkbox: { slotType: View, filter: filterViewProps },
    checkmark: Text,
    content: Text,
  },
  styles: {
    root: [],
    checkbox: [
      backgroundColorTokens,
      borderTokens,
      [
        { source: 'checkboxBackgroundColor', lookup: getPaletteFromTheme, target: 'backgroundColor' },
        { source: 'checkboxBorderColor', lookup: getPaletteFromTheme, target: 'borderColor' },
      ],
    ],
    checkmark: [
      foregroundColorTokens,
      [
        { source: 'checkmarkColor', lookup: getPaletteFromTheme, target: 'color' },
        { source: 'checkmarkVisibility', target: 'opacity' },
      ],
    ],
    content: [foregroundColorTokens, textTokens, [{ source: 'textBorderColor', lookup: getPaletteFromTheme, target: 'borderColor' }]],
  },
});

export default Checkbox;
