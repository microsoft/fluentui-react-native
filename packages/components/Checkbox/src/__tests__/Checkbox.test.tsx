import * as React from 'react';
import type { AccessibilityActionName } from 'react-native';
import { Text, View } from 'react-native';

import type { InteractionEvent } from '@fluentui-react-native/interactive-hooks';
import { checkRenderConsistency, checkReRender } from '@fluentui-react-native/test-tools';
import { Svg } from 'react-native-svg';
import * as renderer from 'react-test-renderer';

import { Checkbox } from '../Checkbox';

function onChange(_e: InteractionEvent, isChecked: boolean) {
  console.log(isChecked);
}

describe('Checkbox component tests', () => {
  it('Checkbox default', () => {
    const tree = renderer.create(<Checkbox label="Default Checkbox" onChange={onChange} defaultChecked={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Checkbox no label', () => {
    const tree = renderer.create(<Checkbox onChange={onChange} defaultChecked={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Checkbox all props', () => {
    const tree = renderer
      .create(
        <Checkbox
          label="All Props Checkbox"
          onChange={onChange}
          defaultChecked={true}
          labelPosition="before"
          disabled
          shape="circular"
          size="large"
          required
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Checkbox all tokens', () => {
    const BoldCheckbox = Checkbox.customize({
      fontFamily: 'Wingdings',
      fontWeight: '900',
      fontSize: 20,
      checkboxBackgroundColor: 'blue',
      checkboxBorderColor: 'red',
      checkboxBorderRadius: 5,
      checkboxBorderWidth: 2,
      checkboxSize: 11,
      checkmarkColor: 'pink',
      checkmarkOpacity: 0.7,
      checkmarkSize: 7,
      backgroundColor: 'purple',
      color: 'yellow',
      spacingLabelAfter: 7,
      spacingLabelBefore: 9,
      borderColor: 'green',
      borderRadius: 4,
      borderWidth: 2,
    });
    const tree = renderer.create(<BoldCheckbox label="All Tokens Checkbox" onChange={onChange} defaultChecked={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Checkbox composed', () => {
    const ComposedCheckbox = Checkbox.compose({
      slots: {
        root: View,
        checkbox: View,
        checkmark: Svg,
        label: Text,
        required: Text,
      },
    });
    const tree = renderer.create(<ComposedCheckbox label="Composed Button with RNText" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Checkbox simple rendering does not invalidate styling', () => {
    checkRenderConsistency(() => <Checkbox>Default button</Checkbox>, 2);
  });

  it('Checkbox re-renders correctly', () => {
    checkReRender(() => <Checkbox>Render twice</Checkbox>, 2);
  });

  it('Checkbox shares produced styles across multiple renders', () => {
    const style = { backgroundColor: 'black' };
    checkRenderConsistency(() => <Checkbox style={style}>Shared styles</Checkbox>, 2);
  });

  it('Checkbox re-renders correctly with style', () => {
    const style = { borderColor: 'blue' };
    checkReRender(() => <Checkbox style={style}>Shared Style Render</Checkbox>, 2);
  });

  it('Checkbox re-renders correctly with accessibilityAction', () => {
    const action = [{ name: 'Expand' as AccessibilityActionName }];
    checkReRender(() => <Checkbox accessibilityActions={action}>Shared Style Render</Checkbox>, 2);
  });
});
