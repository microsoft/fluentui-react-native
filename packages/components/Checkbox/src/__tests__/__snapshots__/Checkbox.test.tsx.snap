// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component tests Checkbox all props 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="All Props Checkbox"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": true,
      "disabled": true,
      "required": true,
    }
  }
  accessible={true}
  defaultChecked={true}
  enableFocusRing={true}
  focusable={false}
  keyUpEvents={
    Array [
      Object {
        "key": " ",
      },
    ]
  }
  labelPosition="before"
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  shape="circular"
  size="large"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": undefined,
      "borderRadius": 4,
      "display": "flex",
      "flexDirection": "row",
      "padding": 8,
      "paddingHorizontal": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      Object {
        "color": "#bdbdbd",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "400",
        "margin": 0,
        "marginEnd": 10,
      }
    }
  >
    All Props Checkbox
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      Object {
        "color": "#690a19",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "400",
        "margin": 0,
        "paddingStart": 8,
      }
    }
  >
    *
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f0f0f0",
        "borderColor": "#e0e0e0",
        "borderRadius": 9999,
        "borderWidth": 1,
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={16}
      bbWidth={16}
      color={4290624957}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": "#bdbdbd",
            "height": 16,
            "opacity": 1,
            "width": 16,
          },
          Object {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      tintColor={4290624957}
      vbHeight={12}
      vbWidth={12}
    >
      <RNSVGGroup
        opacity={1}
      >
        <RNSVGPath
          d="M9.76497 3.20474C10.0661 3.48915 10.0797 3.96383 9.79526 4.26497L5.54526 8.76497C5.40613 8.91228 5.21332 8.99703 5.01071 8.99993C4.8081 9.00282 4.61295 8.92361 4.46967 8.78033L2.21967 6.53033C1.92678 6.23744 1.92678 5.76257 2.21967 5.46967C2.51256 5.17678 2.98744 5.17678 3.28033 5.46967L4.98463 7.17397L8.70474 3.23503C8.98915 2.9339 9.46383 2.92033 9.76497 3.20474Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Checkbox component tests Checkbox all tokens 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="All Tokens Checkbox"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
      "required": false,
    }
  }
  accessible={true}
  defaultChecked={false}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    Array [
      Object {
        "key": " ",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "purple",
      "borderColor": "green",
      "borderRadius": 4,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "padding": 8,
      "paddingHorizontal": 12,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderColor": "red",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={12}
      bbWidth={12}
      color={4294951115}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": "pink",
            "height": 12,
            "opacity": 0.7,
            "width": 12,
          },
          Object {
            "flex": 0,
            "height": 12,
            "width": 12,
          },
        ]
      }
      tintColor={4294951115}
      vbHeight={12}
      vbWidth={12}
    >
      <RNSVGGroup
        opacity={0.7}
      >
        <RNSVGPath
          d="M9.76497 3.20474C10.0661 3.48915 10.0797 3.96383 9.79526 4.26497L5.54526 8.76497C5.40613 8.91228 5.21332 8.99703 5.01071 8.99993C4.8081 9.00282 4.61295 8.92361 4.46967 8.78033L2.21967 6.53033C1.92678 6.23744 1.92678 5.76257 2.21967 5.46967C2.51256 5.17678 2.98744 5.17678 3.28033 5.46967L4.98463 7.17397L8.70474 3.23503C8.98915 2.9339 9.46383 2.92033 9.76497 3.20474Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      Object {
        "color": "yellow",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginEnd": 7,
        "marginStart": 8,
      }
    }
  >
    All Tokens Checkbox
  </Text>
</View>
`;

exports[`Checkbox component tests Checkbox composed 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="Composed Button with RNText"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": undefined,
      "disabled": false,
      "required": false,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    Array [
      Object {
        "key": " ",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onFocus={[Function]}
  onHoverIn={[Function]}
  onHoverOut={[Function]}
  onKeyUp={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": undefined,
      "borderRadius": 4,
      "display": "flex",
      "flexDirection": "row",
      "padding": 8,
      "paddingHorizontal": 12,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#616161",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={12}
      bbWidth={12}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": undefined,
            "height": 12,
            "opacity": 0,
            "width": 12,
          },
          Object {
            "flex": 0,
            "height": 12,
            "width": 12,
          },
        ]
      }
      vbHeight={12}
      vbWidth={12}
    >
      <RNSVGGroup
        opacity={0}
      >
        <RNSVGPath
          d="M9.76497 3.20474C10.0661 3.48915 10.0797 3.96383 9.79526 4.26497L5.54526 8.76497C5.40613 8.91228 5.21332 8.99703 5.01071 8.99993C4.8081 9.00282 4.61295 8.92361 4.46967 8.78033L2.21967 6.53033C1.92678 6.23744 1.92678 5.76257 2.21967 5.46967C2.51256 5.17678 2.98744 5.17678 3.28033 5.46967L4.98463 7.17397L8.70474 3.23503C8.98915 2.9339 9.46383 2.92033 9.76497 3.20474Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Object {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": undefined,
        "lineHeight": undefined,
        "marginStart": 8,
      }
    }
  >
    Composed Button with RNText
  </Text>
</View>
`;

exports[`Checkbox component tests Checkbox default 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="Default Checkbox"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
      "required": false,
    }
  }
  accessible={true}
  defaultChecked={false}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    Array [
      Object {
        "key": " ",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": undefined,
      "borderRadius": 4,
      "display": "flex",
      "flexDirection": "row",
      "padding": 8,
      "paddingHorizontal": 12,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#616161",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={12}
      bbWidth={12}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": undefined,
            "height": 12,
            "opacity": 0,
            "width": 12,
          },
          Object {
            "flex": 0,
            "height": 12,
            "width": 12,
          },
        ]
      }
      vbHeight={12}
      vbWidth={12}
    >
      <RNSVGGroup
        opacity={0}
      >
        <RNSVGPath
          d="M9.76497 3.20474C10.0661 3.48915 10.0797 3.96383 9.79526 4.26497L5.54526 8.76497C5.40613 8.91228 5.21332 8.99703 5.01071 8.99993C4.8081 9.00282 4.61295 8.92361 4.46967 8.78033L2.21967 6.53033C1.92678 6.23744 1.92678 5.76257 2.21967 5.46967C2.51256 5.17678 2.98744 5.17678 3.28033 5.46967L4.98463 7.17397L8.70474 3.23503C8.98915 2.9339 9.46383 2.92033 9.76497 3.20474Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      Object {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginStart": 8,
      }
    }
  >
    Default Checkbox
  </Text>
</View>
`;

exports[`Checkbox component tests Checkbox no label 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
      "required": false,
    }
  }
  accessible={true}
  defaultChecked={false}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    Array [
      Object {
        "key": " ",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": undefined,
      "borderRadius": 2,
      "display": "flex",
      "flexDirection": "row",
      "padding": 4,
      "paddingHorizontal": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#616161",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={12}
      bbWidth={12}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": undefined,
            "height": 12,
            "opacity": 0,
            "width": 12,
          },
          Object {
            "flex": 0,
            "height": 12,
            "width": 12,
          },
        ]
      }
      vbHeight={12}
      vbWidth={12}
    >
      <RNSVGGroup
        opacity={0}
      >
        <RNSVGPath
          d="M9.76497 3.20474C10.0661 3.48915 10.0797 3.96383 9.79526 4.26497L5.54526 8.76497C5.40613 8.91228 5.21332 8.99703 5.01071 8.99993C4.8081 9.00282 4.61295 8.92361 4.46967 8.78033L2.21967 6.53033C1.92678 6.23744 1.92678 5.76257 2.21967 5.46967C2.51256 5.17678 2.98744 5.17678 3.28033 5.46967L4.98463 7.17397L8.70474 3.23503C8.98915 2.9339 9.46383 2.92033 9.76497 3.20474Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      Object {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
      }
    }
  />
</View>
`;
